### ISSUE TITLE
var-naming: allow disabling flags on common initial

### ISSUE THREAD

**Issue by krenzler** [mgechev/revive#1414]
** Is your feature request related to a problem? Please describe. **

I introduced revive as part of my golangci-lint setup to enforce variable naming conventions. However, revive automatically enforces capitalization of common acronyms (like ID, JSON, HTTP) in variable and function names. For example, it insists on getJSON instead of getJson. This strict enforcement causes friction for some projects or teams that prefer to allow lowercase acronyms such as getJson or userId.

Currently, the only way to disable this acronym enforcement is by running revive standalone with a custom revive.toml config, which is not supported when using revive through golangci-lint. This limits flexibility and forces teams either to accept these acronym rules or run revive separately.

 ** Describe the solution you'd like ** 

Please add a configuration option to the var-naming rule—such as disableAcronymEnforcement—that allows users to disable or customize the acronym capitalization checks directly from the YAML config (including when used via golangci-lint). This would:

1. Allow users to disable acronym enforcement while still benefiting from other naming checks.

2. Enable customization of which acronyms are enforced or ignored.

3. Improve revive’s usability and adoption in environments with different naming preferences.

** Describe alternatives you've considered **

1. Running revive standalone with a revive.toml file to disable acronym enforcement, but this is cumbersome and incompatible with golangci-lint.

2. Manually renaming variables to comply with the enforced acronyms, which is not always feasible or desirable, and even native Go itself doesn’t always follow these rules.

3. Trying to align the list of acronyms in my config with revive’s internal list is not feasible.

**Comment by alexandear** [comment]
Thanks for the detailed request!

Just to clarify: it is possible to configure the `revive.var-naming` rule to accept lowercase acronyms like `Id`, `Http`, and `Json` when using `revive` through `golangci-lint`. You can do this by customizing the allow list in your `.golangci.yml` like so:

```yaml
linters:
  settings:
    revive:
      rules:
        - name: var-naming
          arguments:
            - ["ID", "JSON", "HTTP"]
```

Let me know if you run into issues with this approach—happy to help further if needed!

See https://github.com/golangci/golangci-lint/blob/b0c55d28660363573984e9e76214d2136973963b/.golangci.reference.yml#L2744

**Comment by krenzler** [comment]
Thanks for the response.

I'm aware that it's possible to specify initials manually, but in the project I’m working on, we use a large number of common acronyms—including all the ones your project currently recognizes, and quite a few others that are likely to be added in future versions.

That’s why I believe adding the configuration option I suggested earlier could be really helpful. It would make the linter more flexible, allowing teams to flag only the relevant issues without too much overhead.

Let me know what you think—and I’d be happy to help if there’s anything I can contribute :)

**Comment by alexandear** [comment]
Got it.

Your use case makes sense, and adding a configuration option like `disable-acronym-enforcement` (or similar) sounds like a reasonable enhancement.

We’d definitely welcome a contribution if you're interested in implementing this! Feel free to open a PR.

Thanks again for the thoughtful suggestion!
